google_workspace_add_on/src/profile.js

```diff
@@ -0,0 +1,44 @@
+import { PROFILES_KEY } from "./constants";
+import { DeltaSharingClient } from "./delta-sharing-client";
+
+export function getProfilesProperty() {
+    return JSON.parse(PropertiesService.getUserProperties().getProperty(PROFILES_KEY));
+}
+
+export function setProfilesProperty(profiles) {
+    return PropertiesService.getUserProperties().setProperty(
+        PROFILES_KEY, JSON.stringify(profiles));
+}
+
+export function listRedactedProfiles() {
+    let redactedProfiles = getProfilesProperty();
+    for (let [profileName, profile] of Object.entries(redactedProfiles)) {
+        profile.profile.bearerToken = '[REDACTED]';
+    }
+    return redactedProfiles;
+}
+
+export function verifyProfile(profile) {
+    let parsedProfile = JSON.parse(profile);
+    let requiredFields = ['shareCredentialsVersion', 'endpoint', 'bearerToken'];
+    requiredFields.map(field => {
+        if (!(field in parsedProfile)) {
```

Comment:
Best practice is to use `hasOwnProperty` instead of `in`
https://thisthat.dev/has-own-property-vs-in/

Reply:
Done